### ISSUE TITLE
Add support for aws partner eventbus "Event Destinations"

### ISSUE THREAD

**Issue by serle** [stripe/stripe-go#1959]
### Is your feature request related to a problem? Please describe.

The go api currently does not support setting up and configuring aws partner eventbus "Event Destinations". 

### Describe the solution you'd like

I'd like to have a capability similar to the webhook endpoint api, but for aws partner eventbus "Event Destinations".

### Additional context

I would like this to be in place so that I can modify the current terraform provider in order to manage the enabled events via terraform.

**Comment by xavdid-stripe** [comment]
Hi there!

An [API for EventDestinations](https://docs.stripe.com/api/v2/core/event_destinations) was released as part of the `2024-10-28.acacia` API version and supported in the Go SDK as of `v81.0.0`: https://github.com/stripe/stripe-go/releases/tag/v81.0.0. Let us know if that's not working the way you expect though.

Separately, out of curiosity, we'd love to hear what terraform provider you're using. That's very cool!

**Comment by hongyiweiwu01** [comment]
@xavdid-stripe I wasn't able to find this resource in the code (it seems like `EventDestinations` is a `v2` API which the Go SDK broadly still doesn't support?). Could you point to where the resource is located in the Go code? Thank you!

**Comment by jar-stripe** [comment]
Hi @hongyiweiwu01 !  Stepping in for @xavdid-stripe ; you are correct that the Go SDK does not have full v2 support yet, but you can use the Custom requests feature to make calls to v2 APIs and process the results.  See https://github.com/stripe/stripe-go?tab=readme-ov-file#custom-request for an example on how to do this.

**Comment by serle** [comment]
@xavdid-stripe I tried to update [lukasaron/stripe](https://registry.terraform.io/providers/lukasaron/stripe/latest/docs) provider, but since it was using the old terraform v2Sdk I decided to create my own based on the new terraform plugin framework. I've got the stripe provider and a stripe_event_destination resource more or less done. It should be a simple matter to add the other resources as they all follow the same pattern.
